Basket-Donation-Payments/Basket-Donation-Payments/View Controllers/Login & Signup Flow/NonprofitSignupViewController.m

```diff
@@ -61,31 +61,66 @@ -(void)saveNonprofitAndUserToParse {
     self.nonprofit.category = self.nonprofitCategoryTextField.text;
     self.nonprofit.websiteUrlString = self.nonprofitWebsiteTextField.text;
     self.user.nonprofit = self.nonprofit;
-    [self.user signUpInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
-        if (succeeded) {
-            if (self.finishedSavingBoth) {
-                [self performSegueWithIdentifier:@"loginSegue" sender:nil];
-            } else {
-                self.finishedSavingBoth = true;
-            }
+    
+    NSOperationQueue *opQueue = [NSOperationQueue new];
 
-        } else {
-            UIAlertController *alert = [Utils createAlertControllerWithTitle:@"Could not save user." andMessage:error.localizedDescription okCompletion:nil cancelCompletion:nil];
-            [self presentViewController:alert animated:YES completion:nil];
-        }
+    
+    NSBlockOperation *signupUserOp = [NSBlockOperation new];
+    signupUserOp.queuePriority = NSOperationQueuePriorityHigh;
+    __weak NSBlockOperation *weakSignupUserOp = signupUserOp;
```

Comment:
This is unused, but you might need to do `__weak typeof(self) *weakSelf = self;` and use `weakSelf` instead of `self` inside the operation block if you add a strong reference to the operation queue.